lib/logger.js

```diff
@@ -540,13 +570,13 @@ class Logger extends EventEmitter {
     if (has(opts, 'level')) {
       // They've passed in a `level`.  Validate it.
       const level = opts.level.toUpperCase()
-      if (constants.LOG_LEVELS.includes(level)) {
+      if (this[kLogLevels].includes(level)) {
         message.level = level
       } else {
         const err = new Error('Invalid log level.  Using the default instead.')
         err.meta = {
           got: level
-        , expected: constants.LOG_LEVELS
+        , expected: this[kLogLevels]
```

Comment:
You'd want to do `Array.from` or `[...this[kLogLevels]` so you don't pass a direct reference. This leaks an internal and would allow someone to modify it while handling the error.

Reply:
Yep, good call, thanks.

Reply:
Its still passing by reference?

Reply:
It's not really passing anything, just referencing the value for the `meta`.  The code will throw here, so I don't really see leak potential.

Reply:
nvm, this does emit an error.  It throws in the constructor, not here.  Fixing.